src/pal/src/file/directory.cpp

```diff
@@ -267,8 +262,10 @@ RemoveDirectoryW(
         goto done;
     }
 
-    length = (PAL_wcslen(lpPathName)+1) * 3;
+    length = WideCharToMultiByte( CP_ACP, 0, lpPathName, -1, NULL, 0,
+                                   NULL, NULL );
```

Comment:
I'm not sure that this change is necessary. While WideCharToMultiByte gives you an accurate output length, it is much more expensive than wcslen. Since the allocated memory is short-lived, I think it's better to quickly estimate the buffer size using wcslen.